workers/unity/Packages/io.improbable.gdk.debug/WorkerInspector/EntityDetail.cs

```diff
@@ -0,0 +1,138 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Improbable.Gdk.Core;
+using Improbable.Gdk.Debug.WorkerInspector.Codegen;
+using Unity.Entities;
+using UnityEditor;
+using UnityEngine.UIElements;
+
+namespace Improbable.Gdk.Debug.WorkerInspector
+{
+    internal class EntityDetail : VisualElement
+    {
+        private readonly VisualElement componentContainer;
+        private readonly Label entityName;
+        private readonly Label entityId;
+
+        private World world;
+        private EntityData? selected;
+        private readonly List<ComponentVisualElement> visualElements = new List<ComponentVisualElement>();
+
+        public EntityDetail()
+        {
+            const string uxmlPath = "Packages/io.improbable.gdk.debug/WorkerInspector/Templates/EntityDetail.uxml";
+            var template = AssetDatabase.LoadAssetAtPath<VisualTreeAsset>(uxmlPath);
+            template.CloneTree(this);
+
+            entityName = this.Q<Label>("entity-name");
+            entityId = this.Q<Label>("entity-id");
+            componentContainer = this.Q<ScrollView>();
+        }
+
+        public void SetSelectedEntity(EntityData entityData)
+        {
+            selected = entityData;
+            Update();
+        }
+
+        public void SetWorld(World world)
+        {
+            this.world = world;
+            selected = null;
+            componentContainer.Clear();
+            visualElements.Clear();
+            Update();
+        }
+
+        public void Update()
+        {
+            if (!selected.HasValue || !world.EntityManager.Exists(selected.Value.Entity))
+            {
+                selected = null;
+                entityName.text = "No entity selected";
+                entityId.text = string.Empty;
+                return;
+            }
+
+            entityName.text = selected.Value.Metadata;
+            entityId.text = $"Entity ID: {selected.Value.EntityId}";
+
+            UpdateComponentSet();
+
+            foreach (var element in visualElements)
+            {
+                element.Update(world.EntityManager, selected.Value.Entity);
+            }
+        }
+
+        private void UpdateComponentSet()
+        {
+            var components = world.EntityManager.GetComponentTypes(selected.Value.Entity)
```

Comment:
Cant remember, but I guess this returns a managed array?

Reply:
Actually no, its a `NativeArray<>`. I'm surprised the linter doesn't pick this up as needing disposal